src/test/java/org/broadinstitute/hellbender/engine/ProgressMeterUnitTest.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.LongSupplier;
+
+public class ProgressMeterUnitTest extends BaseTest {
+
+    @DataProvider(name = "UpdateIntervalTestData")
+    public Object[][] getUpdateIntervalTestData() {
+        return new Object[][] {
+                // Records per logger update and total number of records to process
+                { 100l, 10000l },
+                { 100l, 10050l },
+                { 100l, 100l },
+                { 100l, 101l },
+                { 100l, 50l },
+                { 100l, 0l }
+        };
+    }
+
+    @Test(dataProvider = "UpdateIntervalTestData")
+    public void testUpdateInterval( final long recordsPerUpdate, final long numRecordsToProcess ) {
+        final ProgressMeter meter = new ProgressMeter(recordsPerUpdate);
+        meter.start();
+        for ( int i = 1; i <= numRecordsToProcess; ++i ) {
+            meter.update(new SimpleInterval("1", 1, 1));
+        }
+        meter.stop();
+
+        // Progress meter should have emitted numRecordsToProcess / recordsPerUpdate status updates.
+        Assert.assertEquals(meter.numLoggerUpdates(), numRecordsToProcess / recordsPerUpdate, "Wrong number of logger updates given recordsPerUpdate = " + recordsPerUpdate);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testInvalidUpdateInterval() {
+        final ProgressMeter meter = new ProgressMeter(0);
+    }
+
+    private static class ListBasedTimeFunction implements LongSupplier {
+        private final List<Long> values;
+        private int currentIndex = 0;
+
+        public ListBasedTimeFunction( final List<Long> values ) {
+            this.values = values;
+        }
+
+        @Override
+        public long getAsLong() {
+            if ( currentIndex >= values.size() ) {
+                throw new NoSuchElementException();
+            }
+
+            return values.get(currentIndex++);
+        }
+
+        public int size() {
+            return values.size();
+        }
+    }
+
+    @DataProvider(name = "ElapsedTimeInMinutesTestData")
+    public Object[][] getElapsedTimeInMinutesTestData() {
+        return new Object[][] {
+                // 600,000ms = 10 minutes
+                { new ListBasedTimeFunction(Arrays.asList(1000l, 601000l)), 10.0 },
```

Comment:
I'd maybe prefer `10 * 60000l + 1000l` for clarity (and same elsewhere, ie factor out the number of minutes and add 1000 separately)